src/triplestore.c

```diff
@@ -6,4 +6,95 @@
 #include "triplestore-redland.c"
 #else
 #include "triplestore-builtin.c"
+#include "dlite-macros.h"
 #endif
+
+
+/*
+  Adds a single triple to store.  The object is considered to be an
+  english literal.  Returns non-zero on error.
+ */
+int triplestore_add_en(TripleStore *ts, const char *s, const char *p,
+                    const char *o)
+{
+  return triplestore_add(ts, s, p, o, "@en");
+}
+
+/*
+  Adds a single triple to store.  The object is considered to be an URI.
+  Returns non-zero on error.
+ */
+int triplestore_add_uri(TripleStore *ts, const char *s, const char *p,
+                        const char *o)
+{
+  return triplestore_add(ts, s, p, o, NULL);
+}
+
+/* Set default namespace */
+void triplestore_set_namespace(TripleStore *ts, const char *ns)
+{
+  if (ts->ns) free((char *)ts->ns);
+  ts->ns = (ns) ? strdup(ns) : NULL;
+}
+
+/* Returns a pointer to default namespace.
+   It may be NULL if it hasn't been set */
+const char *triplestore_get_namespace(TripleStore *ts)
+{
+  return ts->ns;
+}
+
+/*
+  Return pointer to the value for a pair of two criteria.
+
+  Useful if one knows that there may only be one value.  The returned
+  value is held by the triplestore and should be copied by the user
+  since it may be overwritten by later calls to the triplestore.
+
+  Parameters:
+      s, p, o: Criteria to match. Two of these must be non-NULL.
+      d: If not NULL, the required datatype of literal objects.
+      fallback: Value to return if no matches are found.
+      any: If non-zero, return first matching value.
+
+  Returns a pointer to the value of the `s`, `p` or `o` that is NULL.
+  On error NULL is returned.
+ */
```

Comment:
Again, the same documentation in a new place.

Reply:
Yes, we duplicate the documentation of the new `xxx_value()` function a lot. First we document it is triplestore.{c,h} where it is implemented. Then we document it in dlite-collection.{c,h} where it is exposed. Finally in the Python interface. 

I think that it is good to document all functions, even though it results in some duplication. We could strip the documentation from the header files, but that would require to update the doxygen configuration and update the corresponding documentation in the source files. 

Open for other opinions.